osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs

```diff
@@ -47,6 +47,13 @@ public override int CompareTo(FilterCriteria criteria, CarouselItem other)
                     if (ruleset != 0) return ruleset;
 
                     return Beatmap.StarDifficulty.CompareTo(otherBeatmap.Beatmap.StarDifficulty);
+
+                case SortMode.Length:
+                    // Length comparing must be in seconds
```

Comment:
Why though?

Reply:
Just so the user doesn't get confused when comparing with milliseconds, because he can only see length in seconds `m:ss`

Reply:
Not sure why that matters. I don't question why Beatmap A with Length of 4:20 is before Beatmap B with Length of 4:20.

Reply:
> Not sure why that matters. I don't question why Beatmap A with Length of 4:20 is before Beatmap B with Length of 4:20.

it messes up the sorting and makes it weird, that's how it will look like to the end user: https://streamable.com/z6a8y

Reply:
How is it messing up sorting? I don't see anything wrong there.

Reply:
@peppy what do you say, should i compare in seconds or milliseconds, i prefer in seconds

Reply:
answer the question - how is it “messing up”?

Reply:
the user is only able to see the length in seconds, we should sort based on what the length that the user sees, even on stable, if a couple of beatmaps' length **in seconds** are the same, they'll be sorted by difficulty

Reply:
That seems odd as hell. It shouldn't fall back to difficulty randomly. At that point, you may want to consider just *not sorting difficulties* by length.

Reply:
alright